extra/src/main/java/aima/extra/probability/DoubleProbabilityNumber.java

```diff
@@ -309,7 +244,96 @@ public boolean sumsToOne(Iterable<ProbabilityNumber> allProbabilities) {
 		return this.isOne();
 	}
 
+	/**
+	 * Override the precision of ProbabilityNumber instances returned as a
+	 * result of performing operations.
+	 * 
+	 * @param mc
+	 */
+	@Override
+	public void overrideComputationPrecisionGlobally(MathContext mc) {
+		if (mc.getPrecision() > 15) {
+			throw new IllegalArgumentException("Maximum precision possible for DoubleProbabilityNumber is 15");
+		}
+		MAX_PRECISION = mc.getPrecision();
+		ROUNDING_MODE = mc.getRoundingMode();
```

Comment:
This override functionality should be instance based, not class based.  The idea is that the factory/instantiation mechanism put in place will be a good place to override the default to use for all instantiated objects belonging to a particular instance creator.